### DISCUSSION TITLE
Investigate value of precompiled headers

### DISCUSSION THREAD

**Discussion by pajlada** [Chatterino/chatterino2#2725] [Ideas] [answered]
Precompiled headers are currently causing us to have to recompile big parts of the project any time any header file changes, I think it's worth investigating what sort of value we're getting from precompiled headers (i.e. if it speeds up compilations), and if it's worth the tradeoff.

**Comment by fourtf** [comment]
I assume this is only relevant to the pajlada/x headers since they are the only ones in the precompiled header?

**Comment by pajlada** [comment]
I've noticed things being recompiled as soon as any header file is touched, hence the need for investigation

**Comment by fourtf** [comment]
not sure if this is related to precompiled headers but worth looking into it

**Comment by pajlada** [comment]
From my tests, precompiled headers shave off about 70 seconds off the build when using cmake and 4 threads

**Comment by pajlada** [comment]
Currently investigating ccache viability with precompiled headers disabled

**Comment by pajlada** [comment]
Cached builds (using ccache) work properly when precompiled headers are disabled.

I propose adding an option to the CMake call to disable precompiled headers (so they would be enabled by default to keep the current behaviour). 

Thoughts on this @fourtf? The consequence would be adding `#include` directives where missing alongside the CMake option.
If I recall correctly, having `#include`s in files that have already been part of the precompiled header _shouldn't_ slow down the compilation for precompiled headers builds

**Comment by pajlada** [comment, marked as answer]
To add, the original premise of this investigation was false, modifying header files did not cause more file recompiles with precompiled headers than without.
